addons/base_import_module/models/ir_module.py

```diff
@@ -182,6 +187,35 @@ def _import_module(self, module, path, force=False, with_demo=False):
                                 'res_id': attachment.id,
                             })
 
+        # store translation files as attachments to allow loading translations for webclient
+        path_lang = opj(path, 'i18n')
+        if os.path.isdir(path_lang):
+            for entry in os.scandir(path_lang):
+                if not entry.is_file() or not entry.name.endswith('.po'):
+                    # we don't support sub-directories in i18n
+                    continue
+                with file_open(entry.path, 'rb', env=self.env) as fp:
+                    data = base64.b64encode(fp.read())
+                lang = entry.name.split('.')[0]
+                # store as binary ir.attachment
+                values = {
+                    'name': f'{module}_{lang}.po',
+                    'url': f'/{module}/i18n/{lang}.po',
```

Comment:
TBD: maybe there would be a better way to identify the attachment.
e.g. `'model': 'ir.module.module', 'res_id': module.id`